engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -1040,6 +1043,29 @@ struct soul_fragment_t
       dh->resource_gain( RESOURCE_FURY, dh->spec.blind_faith_fury->effectN( 1 ).resource( RESOURCE_FURY ), dh->gain.blind_faith );
     }
 
+    if ( is_type( soul_fragment::LESSER ) && dh->sets->has_set_bonus( DEMON_HUNTER_VENGEANCE, T28, B4))
+    {
+
+     timespan_t duration =  (dh->set_bonuses.t28_vengeance_4pc->effectN( 1 ).time_value()) ;
+
+      if ( dh->cooldown.immolation_aura->is_ready() && !dh->cooldown.fel_devastation->is_ready() )
+      {
+        dh->cooldown.immolation_aura->adjust( -duration );
```

Comment:
Is this (and the next check) not backwards?  This won't really do anything since it's adjusting IA's cooldown only when it is already ready.